src/common/usePreventEventPropagationRef.ts

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+const handleEvent = (event: Event): void => {
+    event.preventDefault();
+    event.stopPropagation();
+};
+
+export default function usePreventEventPropagationRef<T extends HTMLElement>(...args: string[]): (element: T) => void {
```

Comment:
This seems kinda complicated and is largely duplicated in a class component below. Can the mouseup/down handlers be simplified or improved somehow?

Reply:
Hopefully it's mitigated by the fact that the changes in `RegionAnnotations` were removed and pushed down into `PopupReply` as you suggested. But definitely need to look at the mouseup/down handlers to see if it can be simplified somehow in a follow up